build/build_app_engine_package.py

```diff
@@ -15,6 +17,39 @@
 DEST_PATH    Path to the root directory to push/deploy GAE from."""
 
 
+def call_cmd_and_return_output_lines(cmd):
+  process = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+  output = process.communicate()[0]
+  return output.split('\n')
+
+
+def build_version_info_file(dest_path):
+  """Build the version info JSON file."""
+  version_info = {
+      'gitHash': None,
+      'time': None,
+      'branch': None
+  }
+
+  lines = call_cmd_and_return_output_lines(['git', 'log', '-1'])
+  for line in lines:
+    if line.startswith('commit'):
+      version_info['gitHash'] = line.partition(' ')[2].strip()
+    elif line.startswith('Date'):
+      version_info['time'] = line.partition(':')[2].strip()
+    if version_info['gitHash'] is not None and version_info['time'] is not None:
+      break
+
+  lines = call_cmd_and_return_output_lines(['git', 'branch'])
+  for line in lines:
+    if line.startswith('*'):
+      version_info['branch'] = line.partition(' ')[2].strip()
+      break
+
+  with open(dest_path, 'w') as f:
```

Comment:
Done. PTAL

On Fri, Feb 6, 2015 at 5:34 PM, Zeke Chin <notifications@github.com> wrote:

> In build/build_app_engine_package.py
> <https://github.com/webrtc/apprtc/pull/51#discussion_r24284985>:
>
> > +  lines = call_cmd_and_return_output_lines(['git', 'log', '-1'])
> > +  for line in lines:
> > +    if line.startswith('commit'):
> > +      version_info['gitHash'] = line.partition(' ')[2].strip()
> > +    elif line.startswith('Date'):
> > +      version_info['time'] = line.partition(':')[2].strip()
> > +    if version_info['gitHash'] is not None and version_info['time'] is not None:
> > +      break
> > +
> > +  lines = call_cmd_and_return_output_lines(['git', 'branch'])
> > +  for line in lines:
> > +    if line.startswith('*'):
> > +      version_info['branch'] = line.partition(' ')[2].strip()
> > +      break
> > +
> > +  with open(dest_path, 'w') as f:
>
> Safer to either check dest_path exists (even though it does because it's
> being called from main) or handle potential not found exception here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/webrtc/apprtc/pull/51/files#r24284985>.
>



-- 
Jiayang Liu